### ISSUE TITLE
Special Days don't always display

### ISSUE THREAD

**Issue by mirman-mtaggart** [GibbonEdu/core#76]
After adding a few special days successfully, no more will appear on the School Year calendar. Checked the DB for duplicates and none exist. This one's a stumper. Attached are the DB records along with a screenshot showing the invisible days.

v.12.00

![screenshot from 2016-06-20 16-50-52](https://cloud.githubusercontent.com/assets/10470377/16213821/3a9b7296-3707-11e6-9835-3e53ef663577.png)
![screenshot from 2016-06-20 16-50-23](https://cloud.githubusercontent.com/assets/10470377/16213823/3df3b1ce-3707-11e6-83fb-bf7122abb370.png)

**Comment by rossdotparker** [comment]
Thanks for reporting this. Looking at the evidence above, this does look like a bug, although it is not one we have seen in any of our testing or production environments, and we have cases with at least 24 special days.

Are you able to share a dump of your database with support@gibbonedu.org so we can look into this case in more detail?

Thanks,

Ross

**Comment by crayner** [comment]
Hi, Looking at the screen shots provided, I would suggest that you are using special days to define term breaks.  You may need to set your year, then terms.  Special days can then be added as required, to any day in the term, (or not in the term.)  

Please, still provide the information that Ross requested, as we would like the opportunity to fix this issue as well, but using terms then special days may help you get established before we have finished analysing and repairing any issue that you have identified.

Thanks,

Craig 
![image](https://cloud.githubusercontent.com/assets/918953/16215857/0c4daeca-37a9-11e6-9552-d3e37f267b41.png)

**Comment by mirman-mtaggart** [comment]
Ross and Craig,

Thanks for the super-fast response!

To be clear, the term breaks are defined appropriately. We do have long-ish breaks within terms, hence the series of special days.

DB has been sent.

**Comment by rossdotparker** [comment]
Michael, it turns out this was due to a timezone issue. I don't think anyone has set their timezone as far west as LA, and it seems, that once you deviate a certain amount from GMT, it messes with the matching of days based on timestamps. I have fixed this in the v13 commit below:

https://github.com/GibbonEdu/core/commit/b78f64731f06adb3bccd42b9e2b6209e9a758f9a

This commit is backwards compatible to v12, so take function.php and modules/School Admin/schoolYearSpecialDay_manage.php and add them to your v12 install, and hopefully the issue will disappear.

This one two a good 90 minutes to work out ; )
